pkg/metrics/context_metrics.go

```diff
@@ -21,8 +22,27 @@ func MakeContextMetrics() ContextMetrics {
 	return ContextMetrics(make(map[ckey.ContextKey]Metric))
 }
 
+// AddSampleTelemetry counts number of new metrics added.
+type AddSampleTelemetry struct {
+	Total     telemetry.SimpleCounter
```

Comment:
Same question here, as to whether it's necessary to have "total" telemetry or just sum stateful+stateless in the app.